bdk_chain/src/spk_txout_index.rs

```diff
@@ -251,6 +254,26 @@ impl<I: Clone + Ord> SpkTxOutIndex<I> {
         return self.unused.insert(index.clone());
     }
 
+    /// Marks a UTXO(`outpoint`) as selected by adding it to a set of utxos used
+    /// by transactions that have been created but not yet broadcasted.
+    /// Once a utxo is marked as selected by a particular transaction either
+    /// during/after the transaction building process, it cannot be used for
+    /// building another transaction.
+    /// This is useful in a scenario where you are creating a sequence of
+    /// transactions and you don't want utxos to be re-used as inputs
+    /// across transactions.
+    /// A UTXO will be considered selected until it is released by [`Self::unmark_selected()`]
+    /// to the wallet's UTXO pool.
+    pub fn mark_selected(&mut self, outpoint: OutPoint, txid: Txid) -> bool {
```

Comment:
What is the `txid` input for?

Reply:
Just thought maybe it might be helpful to know the tx that reserved a coin. I'm going to remove it.